### PR TITLE
[new release] yaml and yaml-sexp (3.0.1)

### PR DESCRIPTION
**Pull request by avsm**
Parse and generate YAML 1.1/1.2 files

- Project page: <a href="https://github.com/avsm/ocaml-yaml">https://github.com/avsm/ocaml-yaml</a>
- Documentation: <a href="https://avsm.github.io/ocaml-yaml/">https://avsm.github.io/ocaml-yaml/</a>

##### CHANGES:

* Remove dependency on Rresult (@maxtori avsm/ocaml-yaml#50).

### CODE REVIEW

**Comment by avsm** [comment]
cc @kit-ty-kate @hannesm who asked for the release :)

File: packages/yaml/yaml.3.0.1/opam [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+license: "ISC"
+tags: ["org:mirage" "org:ocamllabs"]
+homepage: "https://github.com/avsm/ocaml-yaml"
+doc: "https://avsm.github.io/ocaml-yaml/"
+bug-reports: "https://github.com/avsm/ocaml-yaml/issues"
+depends: [
+  "ocaml" {>= "4.08.0"}
+  "dune" {>= "1.3"}
+  "dune-configurator"
+  "ctypes" {>= "0.12.0"}
```
Comment (mseri) [inline packages/yaml/yaml.3.0.1/opam:28]:
```suggestion
  "ctypes" {>= "0.13.0"}
```

Solves
```
# (cd _build/default && /home/opam/.opam/4.08/bin/ocamlc.opt -w -40 -w -9-27-32 -g -bin-annot -I types/bindings/.yaml_bindings_types.objs -I types/bindings/.yaml_bindings_types.objs/.private -I /home/opam/.opam/4.08/lib/bytes -I /home/opam/.opam/4.08/lib/ctypes -I /home/opam/.opam/4.08/lib/integers -no-alias-deps -o types/bindings/.yaml_bindings_types.objs/yaml_bindings_types.cmi -c -intf types/bindings/yaml_bindings_types.mli)
# File "types/bindings/yaml_bindings_types.mli", line 52, characters 13-24:
# 52 | module M(F : Ctypes.TYPE) : sig
#                   ^^^^^^^^^^^
# Error: Unbound module type Ctypes.TYPE
```

**Comment by mseri** [comment]
There is a failure in `ppx_deriving_yaml` but it looks independent from this PR:
```
#=== ERROR while compiling ppx_deriving_yaml.0.1.0 ============================#
# context              2.0.10 | linux/x86_64 | ocaml-base-compiler.4.13.1 | file:///home/opam/opam-repository
# path                 ~/.opam/4.13/.opam-switch/build/ppx_deriving_yaml.0.1.0
# command              ~/.opam/opam-init/hooks/sandbox.sh build dune build -p ppx_deriving_yaml -j 31 @install @runtest
# exit-code            1
# env-file             ~/.opam/log/ppx_deriving_yaml-8044-60459a.env
# output-file          ~/.opam/log/ppx_deriving_yaml-8044-60459a.out
### output ###
#       ocamlc test/.test.eobjs/byte/dune__exe__Test.{cmi,cmo,cmt} (exit 2)
# (cd _build/default && /home/opam/.opam/4.13/bin/ocamlc.opt -w -40 -g -bin-annot -I test/.test.eobjs/byte -I /home/opam/.opam/4.13/lib/alcotest -I /home/opam/.opam/4.13/lib/alcotest/engine -I /home/opam/.opam/4.13/lib/alcotest/stdlib_ext -I /home/opam/.opam/4.13/lib/astring -I /home/opam/.opam/4.13/lib/bigarray-compat -I /home/opam/.opam/4.13/lib/bytes -I /home/opam/.opam/4.13/lib/cmdliner -I /home/opam/.opam/4.13/lib/ctypes -I /home/opam/.opam/4.13/lib/fmt -I /home/opam/.opam/4.13/lib/integers -I /home/opam/.opam/4.13/lib/re -I /home/opam/.opam/4.13/lib/seq -I /home/opam/.opam/4.13/lib/stdlib-shims -I /home/opam/.opam/4.13/lib/uutf -I /home/opam/.opam/4.13/lib/yaml -I /home/opam/.opam/4.13/lib/yaml/bindings -I /home/opam/.opam/4.13/lib/yaml/bindings/types -I /home/opam/.opam/4.13/lib/yaml/c -I /home/opam/.opam/4.13/lib/yaml/ffi -I /home/opam/.opam/4.13/lib/yaml/types -no-alias-deps -o test/.test.eobjs/byte/dune__exe__Test.cmo -c -impl test/test.pp.ml)
# File "test/test.ml", line 50, characters 0-60:
# 50 | type person = { name : string; age : int } [@@deriving yaml]
#      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
# Error: Unbound module Rresult
# Hint: Did you mean Result?
```
I think the package needs a lower bound on `"rresult" {>= "0.7.0"}`

File: packages/yaml/yaml.3.0.1/opam (hunk 1 above)
Comment (mseri) [reply packages/yaml/yaml.3.0.1/opam]: Can you please upstream this change?

File: packages/yaml/yaml.3.0.1/opam [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+license: "ISC"
+tags: ["org:mirage" "org:ocamllabs"]
+homepage: "https://github.com/avsm/ocaml-yaml"
+doc: "https://avsm.github.io/ocaml-yaml/"
+bug-reports: "https://github.com/avsm/ocaml-yaml/issues"
+depends: [
+  "ocaml" {>= "4.08.0"}
+  "dune" {>= "1.3"}
+  "dune-configurator"
+  "ctypes" {>= "0.13.0"}
```
Comment (mseri) [inline packages/yaml/yaml.3.0.1/opam:28]:
```suggestion
  "ctypes" {>= "0.14.0"}
```

Comment (mseri) [reply packages/yaml/yaml.3.0.1/opam]: For Unbound value Ctypes.CArray.of_string

File: packages/yaml-sexp/yaml-sexp.3.0.1/opam [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+  "ppx_sexp_conv" {>= "v0.9.0"}
+  "sexplib"
+  "yaml" {= version}
+  "mdx" {with-test}
+  "alcotest" {with-test}
+  "crowbar" {with-test}
+  "junit_alcotest" {with-test}
+  "ezjsonm" {with-test}
+  "bos" {with-test}
+]
```
Comment (mseri) [inline packages/yaml-sexp/yaml-sexp.3.0.1/opam:28]:
```suggestion
]
conflicts: [ "result" {<= "1.5"} ]
```

Comment (mseri) [reply packages/yaml-sexp/yaml-sexp.3.0.1/opam:28]: Unbound Result.bind in the lower bounds

File: packages/yaml/yaml.3.0.1/opam [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+  "crowbar" {with-test}
+  "junit_alcotest" {with-test}
+  "ezjsonm" {with-test}
+]
+build: [
+  ["dune" "subst"] {dev}
+  ["dune" "build" "-p" name "-j" jobs]
+  ["dune" "runtest" "-p" name "-j" jobs] {with-test}
+]
+dev-repo: "git+https://github.com/avsm/ocaml-yaml.git"
```
Comment (mseri) [inline packages/yaml/yaml.3.0.1/opam:43]:
```suggestion
conflicts: [ "result" {< "1.5"} ]
dev-repo: "git+https://github.com/avsm/ocaml-yaml.git"
```

**Comment by kit-ty-kate** [comment]
CI was showing:
* [ppx_deriving_yaml.0.1.0 (failed: Unbound module Rresult)](https://opam.ci.ocaml.org/github/ocaml/opam-repository/commit/b8097a60a9e5f285c521cf8286ac4159d61eb1e7/variant/opam-2.1,compilers,4.13,yaml.3.0.1,revdeps,ppx_deriving_yaml.0.1.0)

@mseri could you open a PR to fix this? Also cc @patricoferris

**Comment by mseri** [comment]
Yes, I was already on it. Forgot to link it

**Pull request merged**